pkg/git/gogit/checkout_test.go

```diff
@@ -128,13 +128,21 @@ func TestCheckoutTag_Checkout(t *testing.T) {
 		checkoutTag string
 		expectTag   string
 		expectErr   string
+		setLastRev  bool
 	}{
 		{
 			name:        "Tag",
 			tag:         "tag-1",
 			checkoutTag: "tag-1",
 			expectTag:   "tag-1",
 		},
+		{
+			name:        "Skip Tag if last revision hasn't changed",
+			tag:         "tag-2",
+			checkoutTag: "tag-2",
+			setLastRev:  true,
+			expectErr:   "no changes since last reconcilation",
+		},
```

Comment:
We need an extra test case, which covers when `lastRevision` is set but the revision changed in the repository, and therefore the clone operation goes ahead.